core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -258,6 +258,17 @@ private[spark] class BlockManager(
   @inline final private def isDecommissioning() = {
     decommissioner.isDefined
   }
+
+  @inline final private def checkShouldStore(blockId: BlockId) = {
+    if (isDecommissioning()) {
```

Comment:
Nit: I would avoid the nested if.  
```suggestion
    if (isDecommissioning() && !blockId.isBroadcast) {
```

Reply:
Oh yeah I did the nested if for branch prediction but yeah I'll collapse it down.